src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -78,27 +84,48 @@ public static void apartmentToJson(Apartment a, Path jsonPath) throws IOExceptio
 
 	/**
 	 * Gets the address field from an Address JSON.
+	 * 
+	 * We have to make sure that we have everything at every step <br>
+	 * It is for this reason that the code is filled with check
+	 * <p>
+	 * The contract for this API is returned either <code>"features": []</code> if
+	 * the address cannot be found or in shape <code>{"features": ["properties":
+	 * {"label": "AdressHere"}]}</code>
+	 * </p>
 	 *
-	 * @param jsonString <i>String</i> the Address into JSON format
-	 * @return <i>String</i> the address field
+	 * @param jsonString {@link String} the Address into JSON format
+	 * @return {@link String} the address field
+	 * 
+	 * @throws InvalidObjectException   In case the <code>features</code> field is
+	 *                                  empty
+	 * @throws IllegalArgumentException in the case where the deserialization of the
+	 *                                  JSON encounters a problem or if
+	 *                                  <code>jsonString</code> is blank
 	 */
-	public static String getAddressFromJson(String jsonString) {
-		Jsonb jsonb = JsonbBuilder.create();
-
-		final LinkedHashMap<?, ?> result = jsonb.fromJson(jsonString, LinkedHashMap.class);
-		LOGGER.info("Get address");
-
-		checkArgument(!result.get("address").toString().isEmpty() && result.containsKey("address"),
-				"There is no field adress in the JSON file.");
-		return result.get("address").toString();
+	public static String getAddressFromJson(String jsonString) throws InvalidObjectException {
```

Comment:
Cette méthode ne doit pas se trouver dans cette classe, car son fonctionnement est intimement lié à l’API choisie pour récupérer les adresses.

Reply:
This method is now in `ApartmentFactory`, returns an `Optional<String>` and is `private`